### ISSUE TITLE
buckets to handle fluids in survival

### ISSUE THREAD

**Issue by IntegratedQuantum** [PixelGuys/Cubyz#627]
or maybe something like #473 with an area effect.

**Comment by DellieDelta** [comment]
Did you have any ideas on how buckets could interact with the tool crafting system? Durability on use is the biggest one I think, and maybe how much it is damaged when picking up water vs lava, like that wood holds up really badly with lava

A farther out idea is that bucket capacity is based on the size of the bucket you make, but that the size of the bucket takes away durability

**Comment by Wunka** [comment]
> Did you have any ideas on how buckets could interact with the tool crafting system? Durability on use is the biggest one I think, and maybe how much it is damaged when picking up water vs lava, like that wood holds up really badly with lava

Yes, it would probably contain attributes like flammability and durability. 

> A farther out idea is that bucket capacity is based on the size of the bucket you make, but that the size of the bucket takes away durability

I think capacity is hard to do (but not impossible).

**Comment by DellieDelta** [comment]
Capacity would also need fluid levels, so it might only be something to add later, not necessarily in the first version

The reason I'm asking is because for buckets we have two dependencies. We need a way to only select water/lava when holding a bucket, which #2958 opens up, and we need an onRightClick callback for items, and I saw you were working on something very related in #2898 

Sooo I was thinking I could prepare by setting up the procedural crafting part already :)

**Comment by Wunka** [comment]
> we need an onRightClick callback for items, and I saw you were working on something very related in [#2898](https://github.com/PixelGuys/Cubyz/pull/2898)
> 
> Sooo I was thinking I could prepare by setting up the procedural crafting part already :)

What I am doing is not only related but is exactly that. `onRightClick` would just be an expansion of the system I am writing in #2898

**Comment by DellieDelta** [comment]
Perfectt!

**Comment by IntegratedQuantum** [comment]
> I think capacity is hard to do (but not impossible).

Capacity would be very nice to have, but that will need to wait until we actually figured out how the fluid system should work #62
